File: two-pointers/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
            k = k%nums.length;
            int[] newNums = new int[nums.length];
            int i = 0;
            int diff = nums.length - k;
            for(; i<k && i<nums.length ; i++){
                newNums[i] = nums[diff++];
            }
            int p = 0;
            for(;i<nums.length; i++){
                newNums[i] = nums[p++];
            }
            System.arraycopy(newNums, 0, nums, 0, newNums.length);
    }
}

File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        d = dict()
        l = len(nums)
        for index, value in enumerate(nums):
            d[index] = value
        for i in range(len(nums)):
            nums[(i+k)%l] = d[i]
        
        